service/backend/pchain/backend.go

```diff
@@ -67,24 +67,24 @@ func NewBackend(
 	}
 
 	if nodeMode == service.ModeOnline {
-		if err := backEnd.initChainIDs(); err != nil {
+		if err := b.initChainIDs(); err != nil {
 			return nil, err
 		}
-	}
 
-	backEnd.networkHRP, err = mapper.GetHRP(networkIdentifier)
-	if err != nil {
-		return nil, err
+		if b.networkHRP, err = mapper.GetHRP(b.networkID); err != nil {
```

Comment:
note: we only configure networkHRP correctly if in online mode. Remember that in validateNetworkName we make sure networkID is blank in Offline mode.